q2_quality_control/quality_control.py

```diff
@@ -122,3 +128,115 @@ def evaluate_taxonomy(output_dir: str, expected_taxa: pd.DataFrame,
                misclassifications=None, underclassifications=None,
                composition_regression=None, score_plot=score_plot,
                mismatch_histogram=None, alignments=None)
+
+
+# decontam added code
+_WHOLE_NUM = (lambda x: x >= 0, 'non-negative')
+_PER_NUM = (lambda x: 1 >= x >= 0, 'between 0 and 1')
+_COL_STR = (lambda x: x in {'column_name', 'column_number'},
+            'sample_name or column_name or column_number')
+_DECON_METHOD_STR = (lambda x: x in {'frequency', 'prevalence', 'combined'},
+                     'freqeuncy, prevalence, combined')
+_BOOLEAN = (lambda x: type(x) is bool, 'True or False')
+# Better to choose to skip, than to implicitly ignore things that KeyError
+_SKIP = (lambda x: True, '')
+_valid_inputs = {
+    'asv_or_otu_table': _SKIP,
+    'meta_data': _SKIP,
+    'threshold': _PER_NUM,
+    'decon_method': _DECON_METHOD_STR,
+    'freq_concentration_column': _SKIP,
+    'prev_control_or_exp_sample_column': _SKIP,
+    'prev_control_sample_indicator': _SKIP,
+}
+
+
+def _check_inputs(**kwargs):
+    for param, arg in kwargs.items():
+        check_is_valid, explanation = _valid_inputs[param]
+        if not check_is_valid(arg):
+            raise ValueError('Argument to %r was %r, should be %s.'
+                             % (param, arg, explanation))
+
+
+def _decontam_identify_helper(track_fp, decon_method):
+    df = pd.read_csv(track_fp, sep='\t', index_col=0)
+    df.index.name = '#OTU ID'
+    # removes last column containing true/false information from the dataframe
+    df = df.drop(df.columns[[(len(df.columns)-1)]], axis=1)
```

Comment:
Minor point, the double `[[` isn't needed. Pandas interprets this as indexing with a list of one entry and returns an index of one entry which df.drop() accepts. But with a single bracket, the assumption is that the individual entry is desired, so then a single string is given to df.drop(). (Basically exactly the opposite of R, and that's only a convention some libraries bother with. Everything passes through `__getitem__()` in actuality.)